sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1477,7 +1477,7 @@ pi_result piDeviceRetain(pi_device Device) {
   PI_ASSERT(Device, PI_INVALID_DEVICE);
 
   // The root-device ref-count remains unchanged (always 1).
-  if (Device->IsSubDevice) {
+  if (Device->RootDevice) {
```

Comment:
```suggestion
  if (Device->RootDevice != 0) {
```
Otherwise it looks like the check i for this Device being root-device.
Or rename the field.
Or add IsSubDevice() method.